src/components/base/outline-google-map/src/google-maps-api.ts

```diff
@@ -0,0 +1,140 @@
+import { LitElement } from 'lit';
+import { customElement, property } from 'lit/decorators.js';
+import { ScriptLoaderMap } from './script-loader-map';
+
+abstract class JsonpLibraryElement extends LitElement {
+  public libraryLoaded = false;
+
+  public libraryErrorMessage: string | null = null;
+
+  abstract get libraryUrl(): string;
+
+  abstract get notifyEvent(): string;
+
+  get callbackName(): string | null {
+    return null;
+  }
+
+  private isReady = false;
+
+  libraryUrlChanged() {
+    // can't load before ready because notifyEvent might not be set
+    if (this.isReady && this.libraryUrl != null) this.loadLibrary();
+  }
+
+  libraryLoadCallback(error: Error, detail: object) {
+    if (error) {
+      // eslint-disable-next-line
+      console.warn('Library load failed:', error.message);
+      this.libraryErrorMessage = error.message;
+    } else {
+      this.libraryErrorMessage = null;
+      this.libraryLoaded = true;
+      if (this.notifyEvent != null) {
+        this.dispatchEvent(
+          new CustomEvent(this.notifyEvent, { detail: detail, composed: true })
+        );
+      }
+    }
+  }
+
+  /** loads the library, and fires this.notifyEvent upon completion */
+  loadLibrary() {
+    ScriptLoaderMap.getInstance().require(
+      this.libraryUrl,
+      this.libraryLoadCallback.bind(this),
+      this.callbackName
+    );
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+    this.isReady = true;
+    if (this.libraryUrl != null) this.loadLibrary();
+  }
+}
+// TODO: Resolve ts error: This syntax requires an imported helper but module 'tslib' cannot be found.
+// eslint-disable-next-line
+// @ts-ignore
+@customElement('google-maps-api')
```

Comment:
Question: Is there a benefit of having this be a Lit element rather than something else?

It doesn't appear that anything is rendered here, so maybe this could just be a general module that exports functionality for loading the library?

Reply:
By setting `GoogleMapsApi` as an extension of `JsonpLibraryElement` which itself is an extension of `LitElement` we are getting the lifecycle benefits of a LitElement. We are also wanting to prevent the google maps library to load multiple times. Having this as a component we run the risk of SPA and multiple component use causing memory leaks. This is a method of mitigating those memory leaks. I'm sure there's other ways to achieve these objectives as well.

Reply:
Thanks for the information. :)